### PR TITLE
SELinux compatibility: memalign -> mmap

### PR DESCRIPTION
**Pull request by WPMGPRoSToTeMa**

### CODE REVIEW

File: modules/hamsandwich/hook.h [hunk 1]
```diff
@@ -82,7 +82,9 @@ class Hook
 		ivtable[entry]=(int *)func;
 #if defined(_WIN32)
 		VirtualFree(tramp, 0, MEM_RELEASE);
-#elif defined(__linux__) || defined(__APPLE__)
+#elif defined(__linux__)
+		munmap(tramp, sysconf(_SC_PAGESIZE));
```
Comment (Arkshine) [inline modules/hamsandwich/hook.h]: I might be wrong, but shouldn't you pass the same size as `m_size` value? You will likely need to do something to get such length.

File: public/memtools/CDetour/detourhelpers.h [hunk 2]
```diff
@@ -89,6 +89,8 @@ inline void FreePageMemory(void *addr)
 {
 #if defined(WIN32)
 	VirtualFree(addr, 0, MEM_RELEASE);
+#elif defined(__linux__)
+	munmap(addr, sysconf(_SC_PAGESIZE));
```
Comment (Arkshine) [inline public/memtools/CDetour/detourhelpers.h]: Same here.

File: modules/hamsandwich/hook.h [hunk 3]
```diff
@@ -37,9 +37,10 @@ class Hook
 	int              del;     // 1 if this hook should be destroyed after exec
 	void            *tramp;   // trampoline for this hook
 	char			*ent;     // ent name that's being hooked
+	int              trampSize;
```
Comment (Nextra) [inline modules/hamsandwich/hook.h]: This should also be `size_t`

**Comment by Arkshine** [comment]
Thanks for the changes. This looks technically correct now.

**Pull request merged**